oneargopy/Argo.py

```diff
@@ -1019,3 +1152,300 @@ def __determine_graph_step(self, ax, axis: str)-> int:
+                if parameter in file_variables: 
+                    # We add PRES no matter what, so if the user passed it 
+                    # don't add it to the parameter list at this time.
+                    if parameter != 'PRES' : 
+                        parameter_columns.append(parameter)
+                        parameter_columns.append(parameter + '_QC')
+                        parameter_columns.append(parameter + '_ADJUSTED')
+                        parameter_columns.append(parameter + '_ADJUSTED_QC')
+                        parameter_columns.append(parameter + '_ADJUSTED_ERROR')
+                else : 
+                    print(f'WARNING: The parameter: {parameter} does not exist in the file.')
+            
+            if len(parameter_columns) > 0 : 
+                pressure = ['PRES', 'PRES_QC', 'PRES_ADJUSTED', 'PRES_ADJUSTED_QC', 'PRES_ADJUSTED_ERROR']
+                existing_parameter_columns = pressure + parameter_columns
+                return existing_parameter_columns
+            else : 
+                return None
+        
+        else : 
+            return None
+
+    
+    def __fill_float_data_dataframe(self, files: list)-> pd: 
+        """ A Function to load data into the float data dataframe.
+ 
+            :param: files : list - A list of files to read in data from.
+
+            :return: pd : Dataframe - The dataframe of float data with rows 
+                where measurements were not collected excluded.
+        """
+        if self.download_settings.verbose: print(f'Loading float data...')
+
+        # Getting the file paths for downloaded .nc files
+        directory = Path(self.download_settings.base_dir.joinpath("Profiles"))
+        file_paths = []
+        for file in files : 
+            file_paths.append(directory.joinpath(file))
+
+        # Columns that will always be in the dataframe, these columns are one dimensional
+        static_columns = ['WMOID', 'CYCLE_NUMBER', 'DIRECTION', 
+                                'DATE', 'DATE_QC', 'LATITUDE', 
+                                'LONGITUDE', 'POSITION_QC',]
+        # Columns that need to be calculated or derived 
+        special_case_static_columns = ['DATE', 'DATE_QC', 'WMOID']
+
+        # Empty Dataframe to return at end of function with all loaded data
+        float_data_dataframe = pd.DataFrame()
+
+        # Iterate through files
+        for file in file_paths :             
+            # Open File
+            nc_file = netCDF4.Dataset(file, mode='r')
+            
+            # Get dimensions of .nc file
+            number_of_profiles = nc_file.dimensions['N_PROF'].size
+            number_of_levels = nc_file.dimensions['N_LEVELS'].size
+
+            # Get float id of current file
+            float_id_array = nc_file.variables['PLATFORM_NUMBER'][0]
+            float_id_array = [elem.decode('utf-8') if isinstance(elem, bytes) else elem for elem in float_id_array]
```

Comment:
float_id_array = int(float_id_array.data.tobytes())
should be able to replace lines 1238-1240 (also in at least one other location)

Reply:
float_id = int(float_id_array.data.tobytes().decode('utf-8').strip('\x00'))
has to have a slight change ^ to work but otherwise perfect thanks! much more elegant.